src/pudl/analysis/allocate_net_gen.py

```diff
@@ -622,21 +621,13 @@ def associate_generator_tables(
             on=IDX_PM_ESC,
             how="outer",
             validate="m:1",
-            indicator=True,  # used in _allocate_unassociated_records to find unassocited
         )
         .pipe(remove_inactive_generators)
-        .pipe(
-            _allocate_unassociated_records,
-            idx_cols=IDX_PM_ESC,
-            col_w_unexpected_codes="prime_mover_code",
-            data_columns=[f"{col}_gf_tbl" for col in DATA_COLUMNS],
-        )
-        .drop(columns=["_merge"])  # drop do we can do this again in the bf_summed merge
```

Comment:
applying the treatment in `add_missing_energy_source_codes_to_gens` resulted in 0 records from the gf table (which was merged in just above this chunk) being "unassociated". which is to say all of the records from gf had matching `IDX_PM_ESC` in the updated generators table.

I considered either: a) keeping this in here just in case future records are unassociated OR b) raising an assertion or warning if any records are unassociated/

Reply:
after adding an assertion that failed, i added this step back in. I'm a little confused bc when I was first testing this there seemed to be no unassociated gf data after the new missing-esc treatment. but it definitely seems like there are still unassociated records, but the number dropped dramatically. so i've added the step back in!